Add `Read`, `Write` and `Seek` impls for `Arc<File>` where appropriate

tbu- opened the pull request:
If `&T` implements these traits, `Arc<T>` has no reason not to do so
either. This is useful for operating system handles like `File` or
`TcpStream` which don't need a mutable reference to implement these
traits.

CC #53835.
CC #94744.

rust-highfive commented:
r? @joshtriplett

(rust-highfive has picked a reviewer for you, use r? to override)

tbu- commented:
CC @dtolnay @SNCPlay42

tbu- commented:
Amazing, someone else is doing it. Thanks. :)

tbu- commented:
Hmm, #93044 got closed, so I guess I should reopen this.

tbu- commented:
@rustbot label +T-libs-api -T-libs

tbu- committed:
Forward `io::{Read,Seek,Write}` impls of `File` to `&File`

This reduces code duplication.

tbu- committed:
Add `Read`, `Write` and `Seek` impls for `Arc<File>` where appropriate

If `&T` implements these traits, `Arc<T>` has no reason not to do so
either. This is useful for operating system handles like `File` or
`TcpStream` which don't need a mutable reference to implement these
traits.

CC #53835.
CC #94744.

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-13`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/3809862963/jobs/6481540969) [(plain)](https://github.com/rust-lang/rust/commit/17fa0034a26decad8cb99abc031e7ebd5d04b446/checks/10370854857/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling addr2line v0.17.0
error: function cannot return without recursing
   --> library/std/src/fs.rs:796:5
    |
796 |     fn is_read_vectored(&self) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
797 |         (&*self).is_read_vectored()
    |         --------------------------- recursive call site
    |
    = help: a `loop` may express intention better if this is on purpose
    = note: `-D unconditional-recursion` implied by `-D warnings`
error: function cannot return without recursing
   --> library/std/src/fs.rs:815:5
    |
    |
815 |     fn is_write_vectored(&self) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
816 |         (&*self).is_write_vectored()
    |         ---------------------------- recursive call site
    |
    = help: a `loop` may express intention better if this is on purpose
error: could not compile `std` due to 2 previous errors
Build completed unsuccessfully in 0:00:23
```

</details>

bors commented:
:umbrella: The latest upstream changes (presumably #110655) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

tbu- commented:
@Dylan-DPC What exactly should I change here? I could obviously rebase my PR every half a year while it waits for reviews, but that's just busy work. It'd be nice if this could get reviewed before I rebase it.

ChrisDenton commented:
I think this at least needs someone from @rust-lang/libs-api to sign off on it. Maybe an FCP.

Could you rebase now and use `@rustbot ready` when you're done?

joshtriplett commented:
Since this would be insta-stable, we need an FCP for this:

@rfcbot merge

riking commented on library/std/src/fs.rs:
```diff
@@ -795,67 +796,99 @@ impl Write for File {
     }
+}
 
+#[stable(feature = "io_traits_arc", since = "1.61.0")]
```
Should be CURRENT_RUSTC_VERSION

tbu- replied:
Done.

riking commented on library/std/src/fs.rs:
```diff
@@ -795,67 +796,99 @@ impl Write for File {
+
 #[stable(feature = "rust1", since = "1.0.0")]
-impl Read for &File {
+impl Read for File {
```
Is this change correct wrt the resolver?

tbu- replied:
`Read` (etc.) are still implemented for `File` **and** `&File`. No changes made to the outside. I just moved the implementation into the `&File` impl because I can forward the `File` impl to the `&File` one.

Amanieu commented:
@bors r+

bors commented:
:pushpin: Commit 11fecf619aabd1f3d9a015b925bf0deedc7092e9 has been approved by `Amanieu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Amanieu 11fecf619aabd1f3d9a015b925bf0deedc7092e9 -->
<!-- homu: {"type":"Approved","sha":"11fecf619aabd1f3d9a015b925bf0deedc7092e9","approver":"Amanieu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 11fecf619aabd1f3d9a015b925bf0deedc7092e9 with merge f54adb4db1fbe409eef7632c87ffedfe770dc2d5...
<!-- homu: {"type":"BuildStarted","head_sha":"11fecf619aabd1f3d9a015b925bf0deedc7092e9","merge_sha":"f54adb4db1fbe409eef7632c87ffedfe770dc2d5"} -->

rust-log-analyzer commented:
The job **`i686-mingw`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/5361978923/jobs/9728468946) [(plain)](https://github.com/rust-lang-ci/rust/commit/f54adb4db1fbe409eef7632c87ffedfe770dc2d5/checks/14518058854/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
error: make failed
status: exit code: 2
command: "make"
--- stdout -------------------------------
make[1]: Entering directory '/c/a/rust/rust/tests/run-make/emit-shared-files'
PATH="/c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files:C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\bin:/c/a/rust/rust/build/i686-pc-windows-gnu/stage0-bootstrap-tools/i686-pc-windows-gnu/release/deps:/c/a/rust/rust/build/i686-pc-windows-gnu/stage0/bin:/c/a/rust/rust/ninja:/c/a/rust/rust/mingw32/bin:/c/hostedtoolcache/windows/Python/3.11.4/x64/Scripts:/c/hostedtoolcache/windows/Python/3.11.4/x64:/usr/bin:/c/a/rust/rust/sccache:/c/Program Files/MongoDB/Server/5.0/bin:/c/aliyun-cli:/c/vcpkg:/c/cf-cli:/c/Program Files (x86)/NSIS:/c/tools/zstd:/c/Program Files/Mercurial:/c/hostedtoolcache/windows/stack/2.11.1/x64:/c/cabal/bin:/c/ghcup/bin:/c/Program Files/dotnet:/c/mysql/bin:/c/Program Files/R/R-4.3.0/bin/x64:/c/SeleniumWebDrivers/GeckoDriver:/c/Program Files (x86)/sbt/bin:/c/Program Files (x86)/GitHub CLI:/c/Program Files/Git/bin:/c/Program Files (x86)/pipx_bin:/c/npm/prefix:/c/hostedtoolcache/windows/go/1.20.5/x64/bin:/c/hostedtoolcache/windows/Python/3.7.9/x64/Scripts:/c/hostedtoolcache/windows/Python/3.7.9/x64:/c/hostedtoolcache/windows/Ruby/2.5.9/x64/bin:/c/Program Files/OpenSSL/bin:/c/tools/kotlinc/bin:/c/hostedtoolcache/windows/Java_Temurin-Hotspot_jdk/8.0.372-7/x64/bin:/c/Program Files/ImageMagick-7.1.1-Q16-HDRI:/c/Program Files (x86)/Microsoft SDKs/Azure/CLI2/wbin:/c/ProgramData/kind:/c/Program Files/Eclipse Foundation/jdk-8.0.302.8-hotspot/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/ProgramData/Chocolatey/bin:/c/Program Files/PowerShell/7:/c/Program Files/Microsoft/Web Platform Installer:/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn:/c/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit:/c/Program Files (x86)/Microsoft SQL Server/110/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/120/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/130/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/140/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/150/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/160/DTS/Binn:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/ProgramData/chocolatey/lib/pulumi/tools/Pulumi/bin:/c/Program Files/TortoiseSVN/bin:/c/Program Files/CMake/bin:/c/ProgramData/chocolatey/lib/maven/apache-maven-3.8.7/bin:/c/Program Files/Microsoft Service Fabric/bin/Fabric/Fabric.Code:/c/Program Files/Microsoft SDKs/Service Fabric/Tools/ServiceFabricLocalClusterManager:/c/Program Files/nodejs:/c/Program Files/Git/cmd:/c/Program Files/Git/mingw64/bin:/c/Program Files/Git/usr/bin:/c/Program Files/GitHub CLI:/c/tools/php:/c/Program Files (x86)/sbt/bin:/c/SeleniumWebDrivers/ChromeDriver:/c/SeleniumWebDrivers/EdgeDriver:/c/Program Files/Amazon/AWSCLIV2:/c/Program Files/Amazon/SessionManagerPlugin/bin:/c/Program Files/Amazon/AWSSAMCLI/bin:/c/Program Files (x86)/Google/Cloud SDK/google-cloud-sdk/bin:/c/Program Files (x86)/Microsoft BizTalk Server:/c/Program Files/LLVM/bin:/c/Users/runneradmin/.dotnet/tools:/c/Users/runneradmin/.cargo/bin:/c/Users/runneradmin/AppData/Local/Microsoft/WindowsApps" 'C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\bin\rustdoc.exe' -L C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\lib\rustlib\i686-pc-windows-gnu\lib -Z unstable-options --emit=invocation-specific --output /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only --resource-suffix=-xxx --theme y.css --extend-css z.css x.rs
[ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only/search-index-xxx.js ]
[ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only/settings.html ]
[ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only/x/all.html ]
[ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only/x/index.html ]
[ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only/theme-xxx.css ] # generated from z.css
! [ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only/storage-xxx.js ]
! [ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only/SourceSerif4-It.ttf.woff2 ]
# FIXME: this probably shouldn't have a suffix
[ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only/y-xxx.css ]
# FIXME: this is technically incorrect (see `write_shared`)
! [ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/invocation-only/main-xxx.js ]
PATH="/c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files:C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\bin:/c/a/rust/rust/build/i686-pc-windows-gnu/stage0-bootstrap-tools/i686-pc-windows-gnu/release/deps:/c/a/rust/rust/build/i686-pc-windows-gnu/stage0/bin:/c/a/rust/rust/ninja:/c/a/rust/rust/mingw32/bin:/c/hostedtoolcache/windows/Python/3.11.4/x64/Scripts:/c/hostedtoolcache/windows/Python/3.11.4/x64:/usr/bin:/c/a/rust/rust/sccache:/c/Program Files/MongoDB/Server/5.0/bin:/c/aliyun-cli:/c/vcpkg:/c/cf-cli:/c/Program Files (x86)/NSIS:/c/tools/zstd:/c/Program Files/Mercurial:/c/hostedtoolcache/windows/stack/2.11.1/x64:/c/cabal/bin:/c/ghcup/bin:/c/Program Files/dotnet:/c/mysql/bin:/c/Program Files/R/R-4.3.0/bin/x64:/c/SeleniumWebDrivers/GeckoDriver:/c/Program Files (x86)/sbt/bin:/c/Program Files (x86)/GitHub CLI:/c/Program Files/Git/bin:/c/Program Files (x86)/pipx_bin:/c/npm/prefix:/c/hostedtoolcache/windows/go/1.20.5/x64/bin:/c/hostedtoolcache/windows/Python/3.7.9/x64/Scripts:/c/hostedtoolcache/windows/Python/3.7.9/x64:/c/hostedtoolcache/windows/Ruby/2.5.9/x64/bin:/c/Program Files/OpenSSL/bin:/c/tools/kotlinc/bin:/c/hostedtoolcache/windows/Java_Temurin-Hotspot_jdk/8.0.372-7/x64/bin:/c/Program Files/ImageMagick-7.1.1-Q16-HDRI:/c/Program Files (x86)/Microsoft SDKs/Azure/CLI2/wbin:/c/ProgramData/kind:/c/Program Files/Eclipse Foundation/jdk-8.0.302.8-hotspot/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/ProgramData/Chocolatey/bin:/c/Program Files/PowerShell/7:/c/Program Files/Microsoft/Web Platform Installer:/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn:/c/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit:/c/Program Files (x86)/Microsoft SQL Server/110/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/120/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/130/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/140/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/150/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/160/DTS/Binn:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/ProgramData/chocolatey/lib/pulumi/tools/Pulumi/bin:/c/Program Files/TortoiseSVN/bin:/c/Program Files/CMake/bin:/c/ProgramData/chocolatey/lib/maven/apache-maven-3.8.7/bin:/c/Program Files/Microsoft Service Fabric/bin/Fabric/Fabric.Code:/c/Program Files/Microsoft SDKs/Service Fabric/Tools/ServiceFabricLocalClusterManager:/c/Program Files/nodejs:/c/Program Files/Git/cmd:/c/Program Files/Git/mingw64/bin:/c/Program Files/Git/usr/bin:/c/Program Files/GitHub CLI:/c/tools/php:/c/Program Files (x86)/sbt/bin:/c/SeleniumWebDrivers/ChromeDriver:/c/SeleniumWebDrivers/EdgeDriver:/c/Program Files/Amazon/AWSCLIV2:/c/Program Files/Amazon/SessionManagerPlugin/bin:/c/Program Files/Amazon/AWSSAMCLI/bin:/c/Program Files (x86)/Google/Cloud SDK/google-cloud-sdk/bin:/c/Program Files (x86)/Microsoft BizTalk Server:/c/Program Files/LLVM/bin:/c/Users/runneradmin/.dotnet/tools:/c/Users/runneradmin/.cargo/bin:/c/Users/runneradmin/AppData/Local/Microsoft/WindowsApps" 'C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\bin\rustdoc.exe' -L C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\lib\rustlib\i686-pc-windows-gnu\lib -Z unstable-options --emit=toolchain-shared-resources --output /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/toolchain-only --resource-suffix=-xxx --extend-css z.css x.rs
[ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/toolchain-only/static.files/storage-*.js ]
[ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/toolchain-only/static.files/SourceSerif4-It-*.ttf.woff2 ]
! [ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/toolchain-only/search-index-xxx.js ]
! [ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/toolchain-only/x/index.html ]
! [ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/toolchain-only/theme.css ]
[ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/toolchain-only/static.files/main-*.js ]
! [ -e /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/toolchain-only/y-xxx.css ]
PATH="/c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files:C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\bin:/c/a/rust/rust/build/i686-pc-windows-gnu/stage0-bootstrap-tools/i686-pc-windows-gnu/release/deps:/c/a/rust/rust/build/i686-pc-windows-gnu/stage0/bin:/c/a/rust/rust/ninja:/c/a/rust/rust/mingw32/bin:/c/hostedtoolcache/windows/Python/3.11.4/x64/Scripts:/c/hostedtoolcache/windows/Python/3.11.4/x64:/usr/bin:/c/a/rust/rust/sccache:/c/Program Files/MongoDB/Server/5.0/bin:/c/aliyun-cli:/c/vcpkg:/c/cf-cli:/c/Program Files (x86)/NSIS:/c/tools/zstd:/c/Program Files/Mercurial:/c/hostedtoolcache/windows/stack/2.11.1/x64:/c/cabal/bin:/c/ghcup/bin:/c/Program Files/dotnet:/c/mysql/bin:/c/Program Files/R/R-4.3.0/bin/x64:/c/SeleniumWebDrivers/GeckoDriver:/c/Program Files (x86)/sbt/bin:/c/Program Files (x86)/GitHub CLI:/c/Program Files/Git/bin:/c/Program Files (x86)/pipx_bin:/c/npm/prefix:/c/hostedtoolcache/windows/go/1.20.5/x64/bin:/c/hostedtoolcache/windows/Python/3.7.9/x64/Scripts:/c/hostedtoolcache/windows/Python/3.7.9/x64:/c/hostedtoolcache/windows/Ruby/2.5.9/x64/bin:/c/Program Files/OpenSSL/bin:/c/tools/kotlinc/bin:/c/hostedtoolcache/windows/Java_Temurin-Hotspot_jdk/8.0.372-7/x64/bin:/c/Program Files/ImageMagick-7.1.1-Q16-HDRI:/c/Program Files (x86)/Microsoft SDKs/Azure/CLI2/wbin:/c/ProgramData/kind:/c/Program Files/Eclipse Foundation/jdk-8.0.302.8-hotspot/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/ProgramData/Chocolatey/bin:/c/Program Files/PowerShell/7:/c/Program Files/Microsoft/Web Platform Installer:/c/Program Files/Microsoft SQL Server/130/Tools/Binn:/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn:/c/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit:/c/Program Files (x86)/Microsoft SQL Server/110/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/120/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/130/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/140/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/150/DTS/Binn:/c/Program Files (x86)/Microsoft SQL Server/160/DTS/Binn:/c/Strawberry/c/bin:/c/Strawberry/perl/site/bin:/c/Strawberry/perl/bin:/c/ProgramData/chocolatey/lib/pulumi/tools/Pulumi/bin:/c/Program Files/TortoiseSVN/bin:/c/Program Files/CMake/bin:/c/ProgramData/chocolatey/lib/maven/apache-maven-3.8.7/bin:/c/Program Files/Microsoft Service Fabric/bin/Fabric/Fabric.Code:/c/Program Files/Microsoft SDKs/Service Fabric/Tools/ServiceFabricLocalClusterManager:/c/Program Files/nodejs:/c/Program Files/Git/cmd:/c/Program Files/Git/mingw64/bin:/c/Program Files/Git/usr/bin:/c/Program Files/GitHub CLI:/c/tools/php:/c/Program Files (x86)/sbt/bin:/c/SeleniumWebDrivers/ChromeDriver:/c/SeleniumWebDrivers/EdgeDriver:/c/Program Files/Amazon/AWSCLIV2:/c/Program Files/Amazon/SessionManagerPlugin/bin:/c/Program Files/Amazon/AWSSAMCLI/bin:/c/Program Files (x86)/Google/Cloud SDK/google-cloud-sdk/bin:/c/Program Files (x86)/Microsoft BizTalk Server:/c/Program Files/LLVM/bin:/c/Users/runneradmin/.dotnet/tools:/c/Users/runneradmin/.cargo/bin:/c/Users/runneradmin/AppData/Local/Microsoft/WindowsApps" 'C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\bin\rustdoc.exe' -L C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\lib\rustlib\i686-pc-windows-gnu\lib -Z unstable-options --emit=toolchain-shared-resources,unversioned-shared-resources --output /c/a/rust/rust/build/i686-pc-windows-gnu/test/run-make/emit-shared-files/emit-shared-files/all-shared --resource-suffix=-xxx --extend-css z.css x.rs
make[1]: Leaving directory '/c/a/rust/rust/tests/run-make/emit-shared-files'
--- stderr -------------------------------
--- stderr -------------------------------
warning: theme file "y.css" is missing CSS rules from the default theme
  = warning: the theme may appear incorrect when loaded
  = warning: the theme may appear incorrect when loaded
  = help: to see what rules are missing, call `rustdoc --check-theme "y.css"`

make[1]: *** [Makefile:36: all-shared] Segmentation fault



failures:
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/5361978923/jobs/9729173079)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/5361978923/jobs/9729173079","builder_name":"checks-actions"} -->

fee1-dead commented:
@bors r-

tbu- commented:
Test failure looks unrelated. @rustbot ready

fee1-dead commented:
@bors r=Amanieu

bors commented:
:pushpin: Commit 11fecf619aabd1f3d9a015b925bf0deedc7092e9 has been approved by `Amanieu`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Amanieu 11fecf619aabd1f3d9a015b925bf0deedc7092e9 -->
<!-- homu: {"type":"Approved","sha":"11fecf619aabd1f3d9a015b925bf0deedc7092e9","approver":"Amanieu","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

Mark-Simulacrum commented:
@bors retry

Yielding to stable release.

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/5508671815/jobs/10044138311)
Approved by: Amanieu
Pushing 05b82e551e781154ad2af8a72f34712e97206859 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Amanieu","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/5508671815/jobs/10044138311"},"merge_sha":"05b82e551e781154ad2af8a72f34712e97206859"} -->

heftig commented:
The description and the related PRs promised impls for `Arc<TcpStream>` but it looks like this didn't happen?

yoshuawuyts commented:
@heftig These changes are slated for Rust 1.73, which will release [in two days](https://www.whatrustisit.com).

yoshuawuyts commented:
@heftig yeah you're right; to my knowledge on this PR for `File` has landed and will be part of Rust 1.73. I don't know of any reason why this shouldn't also land for `TcpStream`, so we may want to file a PR for that?
